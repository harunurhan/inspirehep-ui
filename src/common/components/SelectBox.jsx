import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Select } from 'antd';

class SelectBox extends Component {
  constructor(props) {
    super(props);
    this.onChange = this.onChange.bind(this);
  }

  onChange(value) {
    if (this.props.onChange) {
      this.props.onChange(value);
    }
  }

  render() {
    return (
      <Select
        style={{ width: this.props.width }}
        onChange={this.onChange}
        defaultValue={this.props.defaultValue}
      >
        {this.props.options.map(option =>
          <Select.Option key={option.value} value={option.value}>{option.display}</Select.Option>)}
      </Select>
    );
  }
}

SelectBox.propTypes = {
  defaultValue: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.object).isRequired,
  onChange: PropTypes.func,
  width: PropTypes.number,
};

SelectBox.defaultProps = {
  defaultValue: null,
  onChange: null,
  width: 140,
};

export default SelectBox;
