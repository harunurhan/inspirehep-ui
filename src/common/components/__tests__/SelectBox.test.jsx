import React from 'react';
import { shallow } from 'enzyme';

import SelectBox from '../SelectBox';

describe('SelectBox', () => {
  it('render initial state with all props set', () => {
    const options = [
      { value: 'value1', display: 'Value 1' },
      { value: 'value2', display: 'Value 2' },
    ];
    const wrapper = shallow(
      <SelectBox
        defaultValue={options[0].value}
        onChange={jest.fn()}
        options={options}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
