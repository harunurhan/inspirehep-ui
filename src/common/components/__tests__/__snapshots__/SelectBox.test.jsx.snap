// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBox render initial state with all props set 1`] = `
<Select
  choiceTransitionName="zoom"
  defaultValue="value1"
  onChange={[Function]}
  prefixCls="ant-select"
  showSearch={false}
  style={
    Object {
      "width": 140,
    }
  }
  transitionName="slide-up"
>
  <Option
    key="value1"
    value="value1"
  >
    Value 1
  </Option>
  <Option
    key="value2"
    value="value2"
  >
    Value 2
  </Option>
</Select>
`;
